ligo/main.mligo

```diff
@@ -59,7 +59,7 @@ let incr_n_positions (ticks : tick_map) (i : tick_index) (incr : int) =
 
 let collect_fees (s : storage) (key : position_index) : storage * balance_nat =
     let position = match Big_map.find_opt key s.positions with
-    | None -> (failwith "position does not exist" : position_state)
+    | None -> (failwith 102 : position_state)
```

Comment:
After thinking about whether this error is internal or not, I suspect that this error should not exist at all :thinking: 

`collect_fees` is called in `set_position`, and how it is done looks very suspicious.
1. Right before calling `callect_fee` they also check whether position exists, however, there they don't fail if it isn't. I think actually there is reason to fail here too, we can just work with the position as if it was something nullary.
2. `collect_fees` updates storage, however the code that calls `collect_fees` also updates storage _in-parallel_ without accounting for changes in `collect_fees`, that get overwritten after that.

I'm not sure where should we resolve such things, just posting here since it is a bit related.
@dcastro How do you think, where should we consider and fix such bugs?

Reply:
I leave the error as it is for now, and let everyone discuss first whether or not it is a bug.
I'll look into this myself as well.

Reply:
Ah I was just thinking the exact same! :smile: I posted my suspicion on slack: https://serokell.slack.com/archives/C026HTS8DB8/p1629201608004500

So yeah, looks like a legit bug. @rinn7e, can you please open a new ticket for this bug? We'll address it separately.

I agree that this error should be under "Internal Error Codes" and not "Invalid Input Error Codes", but since we're gonna delete it soon it doesn't matter much, I'm ok with leaving it as-is.

Reply:
@rinn7e I've created a ticket to fix this bug here: https://issues.serokell.io/issue/TCFMM-16

Reply:
I've added a `TODO TCFMM-16` there, and leave the error as it is.